src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -52,7 +52,9 @@
     public static void writeReads(
             final JavaSparkContext ctx, final String outputFile, final String referenceFile, final JavaRDD<GATKRead> reads,
             final SAMFileHeader header, ReadsWriteFormat format) throws IOException {
-        writeReads(ctx, outputFile, referenceFile, reads, header, format, 0, null);
+        // NOTE, we must include 'format==ReadsWriteFormat.SINGLE' to preserve the old default behavior for writing spark output
+        // which would not sort the bam if outputting to ReadsWriteFormat.SINGLE. Please use the overload for different sorting behavior.
+        writeReads(ctx, outputFile, referenceFile, reads, header, format, 0, null, true);
```

Comment:
If you think this behavior is important to bwaspark, push it down there.  This is going to bite us in the future if we leave it in.

Reply:
Oh, you did, you just forgot the comment.